infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -374,18 +372,27 @@ private boolean runRecoveryReconfiguration() {
                 localRecoveryLayout.getEpoch(), clusterLayout.getEpoch());
         // The cluster has moved ahead. This node should not force any layout. Let the other
         // members detect that this node has healed and include it in the layout.
-        return clusterLayout.getEpoch() > localRecoveryLayout.getEpoch()
+        boolean recoveryResult = clusterLayout.getEpoch() > localRecoveryLayout.getEpoch()
                 || recoveryReconfigurationResult;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "recoveryResult". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)